storage_mem.go

```diff
@@ -0,0 +1,107 @@
+package main
+
+import (
+	"fmt"
+
+	model "github.com/golang-bristol/beer-model"
+)
+
+// StorageMemory data storage type save only in memory
+type StorageMemory struct {
+	cellar  []model.Beer
+	reviews []model.Review
+}
+
+func newStorageMemory() (*StorageMemory, error) {
+	var err error
+
+	stg := new(StorageMemory)
+
+	return stg, err
+}
+
+// SaveBeer insert or update beers
+func (s *StorageMemory) SaveBeer(beers ...model.Beer) error {
+	for _, beer := range beers {
+		var err error
+
+		beersFound, err := s.FindBeer(model.Beer{ID: beer.ID})
+		if err != nil {
+			return err
+		}
+
+		if len(beersFound) == 1 {
+			*beersFound[0] = beer
+
+		} else if len(beersFound) > 1 {
+			return fmt.Errorf("can't update because found multiple entries with ID %d", beer.ID)
```

Comment:
Completely up to you to decide, but I think it might be better to check for duplicate names rather than ID and auto-assign the next available ID to a beer ourselves rather than take it as a param?

Reply:
This is the same point made for JSON type, the idea for `Save` function is to upsert data, so a form of identifying the record is required to update data when it matches and insert when it doesn't.